gettsim/typing.py

```diff
@@ -21,76 +21,103 @@ def convert_series_to_internal_type(series, internal_type):
     -------
     out : adjusted pd.Series
     """
+    # Copy input series in out
+    out = series.copy()
 
     # Conversion to float
     if internal_type == float:
         # Conversion from boolean to float fails
-        if is_bool_dtype(series):
-            raise ValueError("Conversion of boolean to float is not supported.")
-        elif not is_float_dtype(series):
+        if is_bool_dtype(out):
+            raise ValueError(
+                f"Your variable {series} has the data type boolean."
```

Comment:
`{series}` would print the content of the series, but we should use the name of the column. This will be easy to implement once the errors are collected and only raised in `interface.py`.